### PR TITLE
Allow rendering of deeply nested links

### LINKED ISSUE #1138: Nav: Support nesting of INavLinks to multiple levels deep (tree view)
### Bug Report

- __Package version(s)__: 1.7.2 (but based on the code from master, should repro as well in the latest version)

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: (Blocking, High, Normal, Low): Low (I also need on-demand loading of the children from the server upon expanding the parent, so I ended up writing my own version of the component)

#### Describe the issue:
The INavLinks structure allows each INavLink to have nested children INavLinks. However, when rendering such a tree currently, only the first level of can be expanded and all the other children are never accessible.

#### Actual behavior:
The INavLinkGroups and the first level of INavLinks, if they have any children, are rendered with an expand button in front. Beyond that, none of the children has the expand button, even if the data contains a deep nested structure. 
Seems to be caused by line 127 in Nav.tsx:
        { (nestingLevel === 0 && link.links && link.links.length > 0 ?

#### Expected behavior:
All INavLinks which contain children links have an expand button in front, which can be clicked to show the children

### If applicable, please provide a codepen repro:
http://codepen.io/mihaelams1/pen/ryVXEZ (took the example from here https://fabricreact.azurewebsites.net/fabric-react/master/#/examples/nav as a base and numbered the children to make it clearer). Children 3 & 4 don't get a chance to show up because child 2 doesn't have an expand button.

### PR DESCRIPTION
**Pull request by rrdelaney**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #1138
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This allows deeply nested links groups to render in a `Nav`. It does this by removing the restriction
on `nestingLevel` and adding a `marginLeft` to second-level chevrons.

#### Focus areas to test

Nav component with deeply nest (more than one level deep) links.

### CODE REVIEW

**Comment by micahgodbolt** [comment]
Could you drop in a screenshot demoing how this works/looks?

**Comment by micahgodbolt** [comment]
Also, run `npm run change` and follow the prompts to create a change file, then commit that file to this branch. Thanks for the contribution!

**Comment by rrdelaney** [comment]
@micahgodbolt `npm run change` is telling me:

```
> office-ui-fabric-react-repo@1.0.0 change C:\Users\rydelan.REDMOND\Documents\Source\office-ui-fabric-react
> node node_modules/@microsoft/rush/bin/rush change


Rush Multi-Package Build Tool 3.0.7 - http://aka.ms/rush


No change file is needed.
```

**Comment by rrdelaney** [comment]
Screenshot of change:

localhost-4321-

**Comment by micahgodbolt** [comment]
Oh, it's because you did your work in master. It's trying to compare your current branch to "master", which is itself.

Do you have this repo set as your upstream? If so, make sure you have rush installed `npm install -g @microsoft/rush` and then do `rush change -b upstream`.

**Comment by rrdelaney** [comment]
@micahgodbolt  `rush change -b upstream/master` is still failing for me. The CLI tool only accepts "Enter" the first time I hit it, so I can't answer subsequent questions.

**Comment by micahgodbolt** [comment]
Hmm, okay. We can work around it, but it's certainly something we should take a look at for further contributions.

**Comment by rrdelaney** [comment]
Yea, if I can get rush working I'll let you know.

File: packages/office-ui-fabric-react/src/components/Nav/Nav.tsx [hunk 1]
```diff
@@ -144,8 +144,9 @@ export class Nav extends BaseComponent<INavProps, INavState> implements INav {
           [styles.compositeLinkIsExpanded]: link.isExpanded,
           [styles.compositeLinkIsSelected]: isLinkSelected
         }) }>
-        { (nestingLevel === 0 && link.links && link.links.length > 0 ?
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: you need to include a rush change file; run `npm run change` which should be enough if your origin/master is updated.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: I'm not quite sure what the logic was here for. @inateeg do you have insight on what the nestingLevel check is here?

Comment (rrdelaney) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: @dzearing Rush breaks on my system- I can only submit an answer for the first question. After that it becomes unresponsive.

**Comment by rrdelaney** [comment]
`rush change` wasn't working on my machine in office - I just tried it on my laptop and it worked fine. Commited and pushed the change file 😄

File: packages/office-ui-fabric-react/src/components/Nav/Nav.tsx [hunk 2]
```diff
@@ -144,8 +144,9 @@ export class Nav extends BaseComponent<INavProps, INavState> implements INav {
...
+        { (link.links && link.links.length > 0 ?
           <button
+            style={ { marginLeft: `${14 * nestingLevel}px` } }
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: Not rtl safe...

if you need to dynamically provide left margin, you need to use the getRTL helper to switch this around for rtl. (search the code for examples.)

**Comment by rrdelaney** [comment]
@dzearing Updated for RTL!

**Comment by rrdelaney** [comment]
Any updates on this?

File: packages/office-ui-fabric-react/src/components/Nav/Nav.tsx [hunk 3]
```diff
@@ -135,6 +135,8 @@ export class Nav extends BaseComponent<INavProps, INavState> implements INav {
 
   private _renderCompositeLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {
     const isLinkSelected: boolean = this._isLinkSelected(link);
+    const isRtl: boolean = getRTL();
+    const paddingBefore: string = `${14 * nestingLevel}px`;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: hmm do we have this hardcoded anywhere else? Where did 14 come from? Should that be a const at the top?

Comment (rrdelaney) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: I used `_indentationSize` defined at the top. I don't know why I put 14 instead, but I replaced it with `_indentationSize`.

**Comment by rrdelaney** [comment]
@dzearing I updated to use the constant indentation size defined at the top of the file

**Pull request merged**